src/js/leaflet.storage.js

```diff
@@ -715,6 +742,39 @@ L.Storage.Map.include({
         L.S.fire('ui:start', {data: {html: container}});
     },
 
+    importRaw: function(rawData, type){
+        importedData = JSON.parse(rawData);
+
+        for (option in importedData.properties) {
+            this.options[option] = importedData.properties[option];
+        }
+
+        this.loadOptions();
+
+        var self = this;
+        importedData.layers.forEach( function (geojson) {
+            var dataLayer = self.createDataLayer();
+            dataLayer.fromUmapGeoJSON(geojson);
+        });
+
+        this.save();
+        L.Storage.on('saved', function () {
```

Comment:
You should register the event *before* calling `save`, plus you should use `once` instead of `on`, because we want it to be called only once.